src/game-logic/reducers/updatePeer.js

```diff
@@ -1,18 +1,49 @@
+/**
+ * @typedef {import('../../components/Farmhand/Farmhand').default} Farmhand
+ * @typedef {import('../../components/Farmhand/Farmhand').farmhand.state} farmhand.state
+ * @typedef {import('../../index').farmhand.peerMetadata} farmhand.peerMetadata
+ */
 import { MAX_LATEST_PEER_MESSAGES } from '../../constants'
+import { NEW_COW_OFFERED_FOR_TRADE } from '../../templates'
+import { dialogView } from '../../enums'
+
+import { showNotification } from './showNotification'
 
 /**
  * @param {farmhand.state} state
+ * @param {Farmhand} farmhand
+ * @param {farmhand.peerMetadata} peerMetadata
  * @param {string} peerId The peer to update
- * @param {Object} state
  * @returns {farmhand.state}
  */
-export const updatePeer = (state, peerId, peerState) => {
+export const updatePeer = (state, farmhand, peerMetadata, peerId) => {
   const peers = { ...state.peers }
-  peers[peerId] = peerState
+
+  const previousPeerMetadata = peers[peerId]
+
+  const previousCowOfferedId = previousPeerMetadata?.cowOfferedForTrade?.id
+  const newCowOfferedId = peerMetadata.cowOfferedForTrade?.id
+
+  const isCowNewlyBeingOfferedForTrade =
```

Comment:
this variable is a bit of a mouthful so i wanted to suggest something simpler like `isNewTrade` instead. not a blocking comment by any means though

Reply:
I like it! Done in https://github.com/jeremyckahn/farmhand/commit/d4b00419078efd8fb6f17e2208bd511abf07a795.